brave/brave-core
# Calculate account statement from locally persisted state

Resolves https://github.com/brave/brave-browser/issues/16028

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

- Confirm user is only rewarded if ads are enabled
- Confirm users are rewarded after viewing an ad
- Confirm confirmations are sent to the backend
- Confirm ads are added to 30-day Ads History
- Confirm 30-day Ads History is showing ads for the last 30-days
- Confirm brave://rewards "Current earnings this month (estimated)" has the correct value
- Confirm brave://rewards "Current earnings this month (estimated)" is updated when a user resets Rewards
- Confirm brave://rewards "Next payment date" shows the correct value
- Confirm brave://rewards "Next payment date" is updated when a user resets Rewards
- Confirm brave://rewards "Ads received this month" has the correct value
- Confirm brave://rewards "Ads received this month" is updated when a user resets Rewards
- Confirm Rewards widget "Earnings" has the correct value
- Confirm Rewards widget "Earnings" is updated when a user resets Rewards
- Confirm Rewards panel "Estimated Earnings" has the correct value
- Confirm Rewards panel "Estimated Earnings" is updated when a user resets Rewards
- Confirm Rewards are reset to 0.0 (plus any rewards which were not cashed-in) at midnight on the 1st of each month based on the users local time
- Confirm "Your +#.# BAT payout will begin processing in # day." is working as expected
- Confirm unblinded payment tokens are removed from `confirmations.json` when a user resets Rewards
- Confirm failed confirmations are removed from `confirmations.json` when a user resets Rewards
- Confirm transactions are removed from `database.sqlite` when a user resets Rewards
- Confirm `transactions` are migrated from `confirmations.json` to a new `transactions` table in `database.sqlite` (built from a list of transactions for this month, a single transaction for unredeemed transactions for previous months and a single transaction for redeemed transactions last month)
- Confirm transactions are added to `transactions` table in `database.sqlite`
- Confirm `redeemed_at` column for the 'transactions' table in `database.sqlite` is updated for cashed-out tokens
- Confirm `next_token_redemption_date_in_seconds` is migrated from `confirmations.json` to `brave.brave_ads.rewards.next_time_redemption_at` pref
- Confirm next token redemption date is working as expected
- Confirm day parting frequency cap is working as expected
- Confirm upgrade path from 1.34.x to 1.35.x is working as expected
- Test performance for `transaction` tables that have a large amount of transactions (especially on iOS)

NOTE: `GET /v1/confirmation/payment/{payment_id}` endpoint has been deprecated
